### PR TITLE
#165: Add Method for Update Without Reading Events to Eventuate Aggre…

### PR DESCRIPTION
**Pull request by dartartem**
…gate Store

### CODE REVIEW

File: eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/AggregateCrud.java [hunk 1]
```diff
@@ -14,4 +14,6 @@
   <T extends Aggregate<T>> CompletableFuture<LoadedEvents> find(String aggregateType, String entityId, Optional<AggregateCrudFindOptions> findOptions);
 
   CompletableFuture<EntityIdVersionAndEventIds> update(EntityIdAndType entityIdAndType, Int128 entityVersion, List<EventTypeAndData> events, Optional<AggregateCrudUpdateOptions> updateOptions);
+
+  CompletableFuture<EntityIdVersionAndEventIds> updateWithoutReading(EntityIdAndType entityIdAndType, List<EventTypeAndData> events, Optional<AggregateCrudUpdateWithoutReadingOptions> updateOptions);
```
[the remark this answers is not in the document]
Comment (dartartem) [reply eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/AggregateCrud.java:18]: ok

File: eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/AggregateCrudUpdateWithoutReadingOptions.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package io.eventuate.javaclient.commonimpl.crud;
+
+import io.eventuate.EventContext;
+import org.apache.commons.lang.builder.ToStringBuilder;
+
+import java.util.Optional;
+
+public class AggregateCrudUpdateWithoutReadingOptions {
```
[the remark this answers is not in the document]
Comment (dartartem) [reply eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/AggregateCrudUpdateWithoutReadingOptions.java:8]: Yes, it does not contain unused options.

File: eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/adapters/SyncToAsyncAggregateCrudAdapter.java [hunk 3]
```diff
@@ -4,12 +4,8 @@
 import io.eventuate.CompletableFutureUtil;
 import io.eventuate.EntityIdAndType;
 import io.eventuate.common.id.Int128;
-import io.eventuate.javaclient.commonimpl.crud.AggregateCrudFindOptions;
-import io.eventuate.javaclient.commonimpl.crud.AggregateCrudSaveOptions;
-import io.eventuate.javaclient.commonimpl.crud.AggregateCrudUpdateOptions;
-import io.eventuate.javaclient.commonimpl.crud.EntityIdVersionAndEventIds;
+import io.eventuate.javaclient.commonimpl.crud.*;
```
[the remark this answers is not in the document]
Comment (dartartem) [reply eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/adapters/SyncToAsyncAggregateCrudAdapter.java:7]: It is not me, it is IDEA sometimes optimize imports like that (automatically).

[the remark this answers is not in the document]
Comment (cer) [reply eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/adapters/SyncToAsyncAggregateCrudAdapter.java:7]: That's a setting

File: eventuate-client-java/src/main/java/io/eventuate/UpdateWithoutReadingOptions.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package io.eventuate;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.apache.commons.lang.builder.ToStringBuilder;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class UpdateWithoutReadingOptions {
```
[the remark this answers is not in the document]
Comment (dartartem) [reply eventuate-client-java/src/main/java/io/eventuate/UpdateWithoutReadingOptions.java:10]: I can create a base class. But if new update methods should be renamed to just "update", it is not clear how to design class hierarchy. Probably it is better to remove new option class and just use original class.

File: eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/adapters/SyncToAsyncAggregateCrudAdapter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dartartem) [reply eventuate-client-java-common-impl-crud/src/main/java/io/eventuate/javaclient/commonimpl/crud/adapters/SyncToAsyncAggregateCrudAdapter.java:7]: Ok I will fix the setting

**Pull request merged**